pipeline/pipeline.go

```diff
@@ -106,7 +106,10 @@ func (r *Runtime) Run(runnerCtx context.Context) error {
 	}
 
 	defer func() {
-		if err := r.engine.DestroyWorkflow(runnerCtx, r.spec, r.taskUUID); err != nil {
+		// Use deferCtx because the runnerCtx will be canceled/done in case of error or canceled by user.
```

Comment:
todo: multi tear context -> propagate "shoutdown" mode and still make it possible to hard cancle

Reply:
done #3895 etc ...